Build failed on OpenBSD

## Platform
OS: OPENBSD 7.2 amd64
Compiler: Clang 13.0.0

## Build procedure
```sh
# download & config
☁  compile_from_src  git clone https://github.com/fmtlib/fmt.git
正複製到 'fmt'...
remote: Enumerating objects: 31478, done.
remote: Total 31478 (delta 0), reused 0 (delta 0), pack-reused 31478
接收物件中: 100% (31478/31478), 13.63 MiB | 2.69 MiB/s, 完成.
處理 delta 中: 100% (21322/21322), 完成.
☁  compile_from_src  cd fmt
☁  fmt [master] mkdir BUILD
☁  fmt [master] cd BUILD
☁  BUILD [master] cmake .. -DBUILD_SHARED_LIBS=ON -DFMT_MODULE=ON
-- CMake version: 3.24.2
-- The CXX compiler identification is Clang 13.0.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Module support is disabled.
-- Version: 9.1.1
-- Build type: Release
-- Performing Test HAS_NULLPTR_WARNING
-- Performing Test HAS_NULLPTR_WARNING - Success
-- Target 'doc' disabled (requires doxygen)
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Looking for strptime
-- Looking for strptime - found
-- FMT_PEDANTIC: OFF
-- Configuring done
-- Generating done
-- Build files have been written to: /home/niko/compile_from_src/fmt/BUILD

# build
☁  BUILD [master] cmake --build .
Consolidate compiler generated dependencies of target fmt
[  1%] Building CXX object CMakeFiles/fmt.dir/src/os.cc.o
/home/niko/compile_from_src/fmt/src/os.cc:209:27: error: expected unqualified-id
  int fd = FMT_POSIX_CALL(fileno(file_));
                          ^
/usr/include/stdio.h:410:19: note: expanded from macro 'fileno'
#define fileno(p)       (!__isthreaded ? __sfileno(p) : (fileno)(p))
                        ^
1 error generated.
*** Error 1 in . (CMakeFiles/fmt.dir/build.make:90 'CMakeFiles/fmt.dir/src/os.cc.o': /usr/bin/c++ -DFMT_EXPORT -Dfmt_EXPORTS -I/home/niko/co...)
*** Error 2 in . (CMakeFiles/Makefile2:175 'CMakeFiles/fmt.dir/all': /usr/bin/make -s -f CMakeFiles/fmt.dir/build.make CMakeFiles/fmt.dir/bu...)
*** Error 2 in /home/niko/compile_from_src/fmt/BUILD (Makefile:166 'all': /usr/bin/make -s -f CMakeFiles/Makefile2 all)

```

--- a/src/os.cc
+++ b/src/os.cc
@@ -206,7 +206,11 @@ void buffered_file::close() {
 }
 
 int buffered_file::descriptor() const {
+#ifdef fileno // fileno is a macro on OpenBSD so we cannot use FMT_POSIX_CALL.
+  int fd = fileno(file_);
+#else
   int fd = FMT_POSIX_CALL(fileno(file_));
+#endif
   if (fd == -1)
     FMT_THROW(system_error(errno, FMT_STRING("cannot get file descriptor")));
   return fd;

